data/benchmark/.gitkeep

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
What's the purpose to add this file ?  Actually, I don't know what's the usage of the .gitkeep file.

Reply:
Shall we remove this file? Or even the one with the spark performance tests?

Reply:
The .gitkeep file is to track the empty directory, since a completely empty directory can't be added into Git.

Reply:
I am not sure why we want to create and keep this directory 😢
I have seen that we run the tests like this:
```
./gradlew cleanTest :iceberg-data:jmh -PjmhIncludeRegex=GenericOrcReaderBenchmark -PjmhOutputPath=benchmark/benchmark-result.txt.orc
```

But I am not sure that using the directory is needed for some CI, or other tool, or not

Reply:
Yeah, this seems like a temporary directory that we shouldn't need to have in git.

Reply:
I'm pretty sure we don't need to add this `benchmark` and `benchmark/.gitkeep` to git now.  I just removed this directory and run the jmh command from @pvary , everything is okay and in fact it will just generate the new benchmark directory for the testing report.

@pvary I think we will need to add this `data/benchmark` to the .gitignore file.

Reply:
Removed the `.gitkeep` files, and fixed the `.gitignore` accordingly